pkg/query/columnquery.go

```diff
@@ -732,8 +733,41 @@ func ComputeDiff(ctx context.Context, tracer trace.Tracer, base, compare profile
 	_, span := tracer.Start(ctx, "ComputeDiff")
 	defer span.End()
 
-	records := make([]arrow.Record, 0, len(compare.Samples)+len(base.Samples))
+	var (
+		totalCompareValue          int64
+		totalCompareValuePerSecond float64
+		totalBaseValue             int64
+		totalBaseValuePerSecond    float64
+	)
+	for _, s := range compare.Samples {
+		totalCompareValue += math.Int64.Sum(s.Column(len(s.Columns()) - 4).(*array.Int64))
+		totalCompareValuePerSecond += math.Float64.Sum(s.Column(len(s.Columns()) - 3).(*array.Float64))
+	}
+	for _, s := range base.Samples {
+		totalBaseValue += math.Int64.Sum(s.Column(len(s.Columns()) - 4).(*array.Int64))
+		totalBaseValuePerSecond += math.Float64.Sum(s.Column(len(s.Columns()) - 3).(*array.Float64))
+	}
+
+	// By default, we don't want to scale either side, meaning we multiply by 1.
+	var (
+		cumulativeCompareRatio          = 1.0
+		cumulativeComparePerSecondRatio = 1.0
+		cumulativeBaseRatio             = 1.0
+		cumulativeBasePerSecondRatio    = 1.0
+	)
+	// We figure out which side needs to be scaled based on each profile's total cumulative value.
+	if totalCompareValue > totalBaseValue {
+		cumulativeBaseRatio = float64(totalCompareValue) / float64(totalBaseValue)
+	} else {
+		cumulativeCompareRatio = float64(totalBaseValue) / float64(totalCompareValue)
+	}
+	if totalCompareValuePerSecond > totalBaseValuePerSecond {
```

Comment:
Having two if statements here doesn't make sense to me, that would mean we can have a situation where we're scaling cumulative based on one side and per second on the other. We should always scale the same side.